### PR TITLE
[ShapeOpt] Introducing default gid config

### PR DESCRIPTION
**Pull request by dbaumgaertner**
this avoids unnecessary code duplication with test examples and hides mostly constant settings

### CODE REVIEW

**Review by armingeiser — APPROVED** [review]
🎉 👍

File: applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py [hunk 1]
```diff
@@ -26,43 +26,57 @@
...
+            "gid_configuration" : {
+                "result_file_configuration" : {
+                    "gidpost_flags"         : {
+                        "GiDPostMode"       : "GiD_PostBinary"
+                    },
+                    "output_frequency": 1.0
+                }
+            }
+        }""")
+        self.output_settings["output_format"].RecursivelyValidateAndAssignDefaults(default_gid_settings)
```
Comment (philbucher) [inline applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py]: If you do recursive validation the you cannot add entries to e.g. `gidpost_flags`

Comment (armingeiser) [reply applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py]: I stumbled upon this validation thing already quite some times already.
The missing feature of the parameters is to allow more settings than there are in the defaults, and just compare and validate what is given in the defaults...
Is there another way of doing such a thing?

Comment (philbucher) [reply applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py]: Allowing more things in there would beat the idea of allowing only existing parameters 

The proper solution would be to not add any validation -> the validation should only be done in the class that knows abt the defaults
Basically what you are doing here is redundant that breaks as soon as the parameters in the class change

Comment (armingeiser) [reply applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py]: > The proper solution would be to not add any validation -> the validation should only be done in the class that knows abt the defaults

Thats what i'm saying. A function like `AssignMissingDefaultSettings` that does not do any validation could do the job. We would be able to predefine some top level settings for commonly used Kratos things for the ShapeOptimizationApplication. Then still the class that knows about the defaults will do the proper validation.

> Basically what you are doing here is redundant that breaks as soon as the parameters in the class change

Yes of course, but so do all existing cases....

**Review by armingeiser — CHANGES_REQUESTED** [review]
Some discussions are necessary if this is the place to use `RecursivelyValidateAndAssignDefaults`

File: applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py (hunk 1 above)
Comment (philbucher) [reply applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py]: guess discussing this in person makes more sense :+1:

**Review by armingeiser — APPROVED** [review]

**Pull request merged**